src/components/application_manager/test/application_manager_impl_test.cc

```diff
@@ -35,78 +35,236 @@
 const uint32_t kStopStreamingTimeout = 10000;
+const uint32_t kWaitFreeQueue = 2000;
+const std::string directory_name = "./test_storage";
+
+bool is_called_ = false;
+uint32_t hmi_app_id_ = 0u;
+sync_primitives::Lock state_lock_;
+sync_primitives::ConditionalVariable state_condition_;
 }  // namespace
 
 class ApplicationManagerImplTest : public ::testing::Test {
  protected:
-  virtual void SetUp() OVERRIDE {
-    app_manager_impl.reset(new am::ApplicationManagerImpl(
+  void SetUp() OVERRIDE {
+    app_id_ = 10;
+
+    CreateAppManager();
+
+    const std::string kMacAddr_("kMacAddr_ess");
+
+    ON_CALL(mock_session_observer, GetDataOnDeviceID(_, _, _, _, _))
+        .WillByDefault(DoAll(SetArgPointee<3u>(kMacAddr_), Return(0)));
+    ON_CALL(mock_connection_handler, GetDataOnSessionKey(_, _, _, _))
+        .WillByDefault(DoAll(SetArgPointee<3u>(app_id_), Return(0)));
+    ON_CALL(mock_connection_handler, get_session_observer())
+        .WillByDefault(ReturnRef(mock_session_observer));
+
+    mock_storage =
+        ::utils::MakeShared<NiceMock<resumption_test::MockResumptionData>>(
+            *app_manager_impl_);
+
+    app_manager_impl_->resume_controller().set_resumption_storage(mock_storage);
+    app_manager_impl_->set_connection_handler(&mock_connection_handler);
+    app_manager_impl_->set_hmi_message_handler(&mock_hmi_mess_handler);
+  }
+
+  void CreateAppManager() {
+    const uint32_t timeout = 10000u;
+    const std::string folder_name = "app";
+    const std::string test_path =
+        file_system::ConcatPath(directory_name, folder_name);
+    const std::string file1 =
+        file_system::ConcatPath(test_path, "test_file2.bmp");
+
+    ASSERT_TRUE(file_system::CreateDirectoryRecursively(test_path));
+    ASSERT_TRUE(file_system::CreateFile(file1));
+
+    ON_CALL(mock_application_manager_settings_, thread_pool_size())
+        .WillByDefault(Return(2u));
+    ON_CALL(mock_application_manager_settings_, app_icons_folder())
+        .WillByDefault(ReturnRef(directory_name));
+    ON_CALL(mock_application_manager_settings_, app_storage_folder())
+        .WillByDefault(ReturnRef(directory_name));
+    ON_CALL(mock_application_manager_settings_, launch_hmi())
+        .WillByDefault(Return(true));
+    ON_CALL(mock_application_manager_settings_, stop_streaming_timeout())
+        .WillByDefault(Return(1u));
+    ON_CALL(mock_application_manager_settings_, default_timeout())
+        .WillByDefault(ReturnRef(timeout));
+    app_manager_impl_.reset(new am::ApplicationManagerImpl(
         mock_application_manager_settings_, mock_policy_settings_));
 
-    app_id_ = 10;
+    ASSERT_TRUE(app_manager_impl_.get());
   }
 
   uint32_t app_id_;
-  MockApplicationManagerSettings mock_application_manager_settings_;
-  policy_test::MockPolicySettings mock_policy_settings_;
+  NiceMock<policy_test::MockPolicySettings> mock_policy_settings_;
+  utils::SharedPtr<NiceMock<resumption_test::MockResumptionData>> mock_storage;
```

Comment:
@AGritsevich Nothing is done according my comment in a22bce6cd3374728eb2b28e82b02bf72f8682fb9